default.nix

```diff
@@ -50,6 +50,7 @@ stdenv.mkDerivation {
     menhirLib
     yojson
     zarith
+    merlin
```

Comment:
Can we move this to `shell.nix`, as a dev environment dependency?

Reply:
Yes, but I would like to be sure to have merlin when I start nix-shell. 
I did it like this, because I don't know how to do it in the right way.

Reply:
Help is welcome

Reply:
I tried something, but Francois will have to confirm that.

Reply:
Thanks, I was on my phone last night and was planning on doing this this morning.

It is what I had in mind (although I was going to do it as a separate PR). @bgregoir, can you check that it gives you the development environment you want for now?

We should discuss the right way of offering support for various IDEs and all, but let's get this working for core devs for now.

Reply:
@bgregoir feel free to resolve this conversation if it works for you. Once it's resolved, this is fine for me (but my review of the logic itself is limited).